### PR TITLE
Extend tests

### PR DESCRIPTION
**Pull request by ejoerns**
This is some pre-work for further test setups that fixes some issues, reduces a little bit of code duplication and extend tests by some newly added one and some more tested details.

* Adds test for some error paths in bundle checks
* shares common code in bundle test routines
* Tests if updates actually write content
* Add a little bit of test description

### CODE REVIEW

File: src/install.c [hunk 1]
```diff
@@ -1127,11 +1127,10 @@ gboolean do_install_network(const gchar *url, GError **error) {
 	if (r_context()->config->preinstall_handler) {
 		g_print("Starting pre install handler: %s\n", r_context()->config->preinstall_handler);
 		res = launch_and_wait_handler(base_url, r_context()->config->preinstall_handler, manifest, target_group, &ierror);
-	}
-
-	if (!res) {
-		g_propagate_prefixed_error(error, ierror, "Handler error: ");
-		goto out;
+		if (!res) {
+			g_print("Handler error: %s\n", r_context()->config->preinstall_handler);
```
Comment (jluebbe) [inline src/install.c]: This should stay g_propagate_prefixed_error (also below).

File: test/bundle.c [hunk 2]
```diff
@@ -36,6 +36,42 @@ static void bundle_fixture_tear_down(BundleFixture *fixture,
...
 }
 
+static void test_check_empty_bundle(BundleFixture *fixture,
+		gconstpointer user_data)
+{
+	gchar *bundlename;
+	gsize size;
+	GError *ierror = NULL;
+
+	random_bytes(1234, 10);
```
Comment (jluebbe) [inline test/bundle.c]: Remove.

File: test/common.c [hunk 3]
```diff
@@ -14,6 +14,32 @@ typedef struct {
...
+			    gsize size, const guint32 seed) {
+	gchar *pathname;
+	gchar *content;
+
+	pathname = g_build_filename(tmpdir, filename, NULL);
+	g_assert_nonnull(pathname);
+
+	content = random_bytes(size, seed);
+
+	if (!g_file_set_contents(pathname, content, -1, NULL)) {
```
Comment (jluebbe) [inline test/common.c]: Explicitly pass the size, content may contain zero bytes.

**Comment by ejoerns** [comment]
``` diff
diff --git a/src/install.c b/src/install.c
index 82201da..0d6bbfc 100644
--- a/src/install.c
+++ b/src/install.c
@@ -1021,12 +1021,12 @@ gboolean do_install_bundle(RaucInstallArgs *args, GError **error) {
        if (r_context()->config->preinstall_handler) {
                g_print("Starting pre install handler: %s\n", r_context()->config->preinstall_handler);
                res = launch_and_wait_handler(mountpoint, r_context()->config->preinstall_handler, manifest, target_group, &ierror);
+               if (!res) {
+                       g_propagate_prefixed_error(error, ierror, "Handler error: ");
+                       goto out;
+               }
        }
 
-       if (!res) {
-               g_propagate_prefixed_error(error, ierror, "Handler error: ");
-               goto out;
-       }
 
        if (manifest->handler_name) {
                g_print("Using custom handler: %s\n", manifest->handler_name);
@@ -1044,12 +1044,12 @@ gboolean do_install_bundle(RaucInstallArgs *args, GError **error) {
        if (r_context()->config->postinstall_handler) {
                g_print("Starting post install handler: %s\n", r_context()->config->postinstall_handler);
                res = launch_and_wait_handler(mountpoint, r_context()->config->postinstall_handler, manifest, target_group, &ierror);
+               if (!res) {
+                       g_propagate_prefixed_error(error, ierror, "Handler error: ");
+                       goto umount;
+               }
        }
 
-       if (!res) {
-               g_propagate_prefixed_error(error, ierror, "Handler error: ");
-               goto umount;
-       }
 
        res = TRUE;
 
@@ -1128,11 +1128,12 @@ gboolean do_install_network(const gchar *url, GError **error) {
                g_print("Starting pre install handler: %s\n", r_context()->config->preinstall_handler);
                res = launch_and_wait_handler(base_url, r_context()->config->preinstall_handler, manifest, target_group, &ierror);
                if (!res) {
-                       g_print("Handler error: %s\n", r_context()->config->preinstall_handler);
+                       g_propagate_prefixed_error(error, ierror, "Handler error: ");
                        goto out;
                }
        }
 
+
        g_print("Using network handler for %s\n", base_url);
        res = launch_and_wait_network_handler(base_url, manifest, target_group);
        if (!res) {
@@ -1145,11 +1146,12 @@ gboolean do_install_network(const gchar *url, GError **error) {
                g_print("Starting post install handler: %s\n", r_context()->config->postinstall_handler);
                res = launch_and_wait_handler(base_url, r_context()->config->postinstall_handler, manifest, target_group, &ierror);
                if (!res) {
-                       g_print("Handler error: %s\n", r_context()->config->postinstall_handler);
+                       g_propagate_prefixed_error(error, ierror, "Handler error: ");
                        goto out;
                }
        }
 
+
        res = TRUE;
 
 out:
diff --git a/test/bundle.c b/test/bundle.c
index 42213df..0ce5778 100644
--- a/test/bundle.c
+++ b/test/bundle.c
@@ -2,6 +2,7 @@
 #include <locale.h>
 #include <glib.h>
 #include <glib/gstdio.h>
+#include <gio/gio.h>
 
 #include <bundle.h>
 #include <context.h>
@@ -42,15 +43,14 @@ static void test_check_empty_bundle(BundleFixture *fixture,
        gchar *bundlename;
        gsize size;
        GError *ierror = NULL;
-
-       random_bytes(1234, 10);
+       gboolean res = FALSE;
 
        bundlename = write_random_file(fixture->tmpdir, "bundle.raucb", 0, 1234);
        g_assert_nonnull(bundlename);
 
-       g_assert_false(check_bundle(bundlename, &size, TRUE, &ierror));
-       g_assert_nonnull(ierror->message);
-       g_assert_cmpstr(ierror->message, ==, "failed to seek to end of bundle: Error seeking in file: Invalid argument");
+       res = check_bundle(bundlename, &size, TRUE, &ierror);
+       g_assert_error(ierror, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT);
+       g_assert_false(res);
 
        g_free(bundlename);
 }
@@ -61,13 +61,14 @@ static void test_check_invalid_bundle(BundleFixture *fixture,
        gchar *bundlename;
        gsize size;
        GError *ierror = NULL;
+       gboolean res = FALSE;
 
        bundlename = write_random_file(fixture->tmpdir, "bundle.raucb", 1024, 1234);
        g_assert_nonnull(bundlename);
 
-       g_assert_false(check_bundle(bundlename, &size, FALSE, &ierror));
-       g_assert_nonnull(ierror->message);
-       g_assert_cmpstr(ierror->message, ==, "failed to seek to start of bundle signature: Error seeking in file: Invalid argument");
+       res = check_bundle(bundlename, &size, FALSE, &ierror);
+       g_assert_error(ierror, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT);
+       g_assert_false(res);
 
        g_free(bundlename);
 }
diff --git a/test/common.c b/test/common.c
index 8b4ee1f..497700e 100644
--- a/test/common.c
+++ b/test/common.c
@@ -33,7 +33,7 @@ gchar* write_random_file(const gchar *tmpdir, const gchar *filename,
 
        content = random_bytes(size, seed);
 
-       if (!g_file_set_contents(pathname, content, -1, NULL)) {
+       if (!g_file_set_contents(pathname, content, size, NULL)) {
                return NULL;
        }
 

```

**Comment by ejoerns** [comment]
```diff
diff --git a/include/bundle.h b/include/bundle.h
index b766c70..5503dcf 100644
--- a/include/bundle.h
+++ b/include/bundle.h
@@ -2,6 +2,13 @@
 
 #include <glib.h>
 
+#define R_BUNDLE_ERROR r_bundle_error_quark ()
+GQuark r_bundle_error_quark(void);
+
+typedef enum {
+       R_BUNDLE_ERROR_SIGNATURE
+} RBundleError;
+
 gboolean create_bundle(const gchar *bundlename, const gchar *contentdir, GError **error);
 gboolean check_bundle(const gchar *bundlename, gsize *size, gboolean verify, GError **error);
 gboolean extract_bundle(const gchar *bundlename, const gchar *outputdir, gboolean verify, GError **error);
diff --git a/src/bundle.c b/src/bundle.c
index 72e25a7..e5770ef 100644
--- a/src/bundle.c
+++ b/src/bundle.c
@@ -6,6 +6,12 @@
 #include <mount.h>
 #include "bundle.h"
 
+GQuark
+r_bundle_error_quark (void)
+{
+  return g_quark_from_static_string ("r-bundle-error-quark");
+}
+
 static gboolean mksquashfs(const gchar *bundlename, const gchar *contentdir, GError **error) {
        GSubprocess *sproc = NULL;
        GError *ierror = NULL;
@@ -289,6 +295,21 @@ gboolean check_bundle(const gchar *bundlename, gsize *size, gboolean verify, GEr
                goto out;
        }
 
+       /* sanity check: signature should be smaller than bundle size */
+       if (sigsize > (guint64)offset) {
+               g_set_error(error, R_BUNDLE_ERROR, R_BUNDLE_ERROR_SIGNATURE,
+                               "signature size sanity check failed: %"G_GSIZE_FORMAT" exceeds bundle size", sigsize);
+               res = FALSE;
+               goto out;
+       }
+       /* sanity check: signature should be smaller than 64kiB */
+       if (sigsize > 0x4000000) {
+               g_set_error(error, R_BUNDLE_ERROR, R_BUNDLE_ERROR_SIGNATURE,
+                               "signature size sanity check failed: %"G_GSIZE_FORMAT" exceeds 64KiB", sigsize);
+               res = FALSE;
+               goto out;
+       }
+
        offset -= sigsize;
 
        if (size)
diff --git a/test/bundle.c b/test/bundle.c
index 0ce5778..2b42a68 100644
--- a/test/bundle.c
+++ b/test/bundle.c
@@ -67,7 +67,7 @@ static void test_check_invalid_bundle(BundleFixture *fixture,
        g_assert_nonnull(bundlename);
 
        res = check_bundle(bundlename, &size, FALSE, &ierror);
-       g_assert_error(ierror, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT);
+       g_assert_error(ierror, R_BUNDLE_ERROR, R_BUNDLE_ERROR_SIGNATURE);
        g_assert_false(res);
 
        g_free(bundlename);
diff --git a/test/common.c b/test/common.c
index 497700e..c0e6768 100644
--- a/test/common.c
+++ b/test/common.c
@@ -37,6 +37,7 @@ gchar* write_random_file(const gchar *tmpdir, const gchar *filename,
                return NULL;
        }
 
+       g_free(content);
        return pathname;
 }
 

```

**Pull request merged**